astropy/utils/data.py

```diff
@@ -1164,6 +1257,29 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None,
         ``User-Agent: some_value`` and ``Accept: */*``, where ``some_value``
         is set by ``astropy.utils.data.conf.default_http_user_agent``.
 
+    ssl_context : dict, optional
+        Keyword arguments to pass to `ssl.create_default_context` when
+        downloading from HTTPS or TLS+FTP sources.  This can be used provide
+        alternative paths to root CA certificates.  Additionally, if the key
+        ``'certfile'`` and optionally ``'keyfile'`` and ``'password'`` are
+        included, they are passed to passed to a call to
+        `ssl.SSLContext.load_cert_chain`.  This can be used for performing
+        SSL/TLS client certificate authentication.
+
+    allow_insecure : bool, optional
+        Allow downloading files over a TLS/SSL connection even when the server
+        certificate verification failed.  When set to `True` the potentially
+        insecure download is allowed to proceed, but an
+        `~astropy.utils.exceptions.AstropyWarning` is issued.  The default
+        value for this can be set by
+        ``astropy.utils.data.conf.default_allow_insecure``.  If you are getting
```

Comment:
If we agree to remove the config item, this can go too.
```suggestion
        `~astropy.utils.exceptions.AstropyWarning` is issued.
        If you are getting
```